src/downshift.js

```diff
@@ -622,6 +622,11 @@ class Downshift extends Component {
         inputValue: this.props.itemToString(this.props.selectedItem),
       })
     }
+
+    if (this.props.highlightedIndex !== prevProps.highlightedIndex) {
```

Comment:
We should probably add a check for whether `highlightedIndex` is a state prop as well (like we do with `selectedItem` above).